server/manager/src/main/java/org/apache/accumulo/manager/Manager.java

```diff
@@ -1265,14 +1265,21 @@ public void run() {
 
     context.getTableManager().addObserver(this);
 
-    Thread statusThread = Threads.createThread("Status Thread", new StatusThread());
+    // TODO KEVIN RATHBUN updating the Manager state seems like a critical function. However, the
+    // thread already handles, waits, and continues in the case of any Exception, so critical or
+    // non critical doesn't make a difference here.
+    Thread statusThread = Threads.createCriticalThread("Status Thread", new StatusThread());
     statusThread.start();
 
-    Threads.createThread("Migration Cleanup Thread", new MigrationCleanupThread()).start();
+    // TODO KEVIN RATHBUN migration cleanup may be a critical function of the manager, but the
+    // thread will already handle, wait, and continue in the case of any Exception, so critical
+    // or non critical doesn't make a difference here.
+    Threads.createCriticalThread("Migration Cleanup Thread", new MigrationCleanupThread()).start();
 
     tserverSet.startListeningForTabletServerChanges();
 
-    Threads.createThread("ScanServer Cleanup Thread", new ScanServerZKCleaner()).start();
+    // TODO KEVIN RATHBUN Some ZK cleanup doesn't seem like a critical function of manager
+    Threads.createNonCriticalThread("ScanServer Cleanup Thread", new ScanServerZKCleaner()).start();
```

Comment:
Some ZK cleanup doesn't seem like a critical function of manager

Reply:
I'm thinking we may want to make this critical.  The clients find ScanServers by looking them up in ZooKeeper. Leaving orphaned entries over a long period of time could progressively slow the clients down. I'm not sure why the thread might die, as all of the known exceptions are handled, but RuntimeException is not caught by the thread, so some unchecked exception could cause it to fail.

Reply:
Addressed in 2c4ce6bc57b166a2a57eb61c11df5b2bb8da76b8